code/modules/client/preference/preferences_mysql.dm

```diff
@@ -90,7 +90,7 @@
 					UI_style_alpha='[UI_style_alpha]',
 					be_role='[sanitizeSQL(list2params(be_special))]',
 					default_slot='[default_slot]',
-					toggles='[num2text(toggles, 7)]',
+					toggles='[num2text(toggles, 9)]',
```

Comment:
I wonder what this number means 🤔 

Is there a suitable define that could be used?

Reply:
Number is significant figures so if you have a calculation to get the sigfigs of any bit let me know

Reply:
You can do this with ceiling(log10(n))

Reply:
I'm not sure if `log10` directly exists, you'll want `log(MAX_TOGGLES, 10)`
And I'm not sure if `ceiling` directly exists either
An analogue for it would be `round(log(MAX_TOGGLES, 10)) + 1` (since round does a `floor`, or the number below the current one)

Reply:
how about `Ceiling(log(10, (TOGGLES_TOTAL+1))`

Reply:
And, the reason why:
the `log10(x)` operation, gets the value `n` for which `10**n = x`
Put in normal people speak, it's how many times you need to multiply 10 by itself to get `x`
So, say, 240 - `log10(240) = 2.38`. Thus, `240 = 10**(2 + .38) = 10**2 * 10**.38 = 100 * 2.4.
And since the decimal system is based around powers of 10, we can get a convenient method of getting the number of digits in a decimal value by using log10 and rounding up to the nearest 1.

Reply:
Awesome, this means we won't have to keep changing this in the future